### PR TITLE
Geometric policy blending on cuda backend

### PR DESCRIPTION
**Pull request by Kovax007**
Geometric blending means a weighted sum of the heads' logits, taken before the softmax. It returns a single policy vector that nothing downstream can distinguish from an unblended one. PolicyTemperature is applied identically to every head, so it factors out of the mixture entirely and keeps its usual meaning.

Weights are set as backend options, and the heads that carry weight are decided once when the backend is built:

    --backend-opts=policy_blend_optimistic=0.55
    --backend-opts=policy_blend_optimistic=0.55,policy_blend_soft=0.1

The two also exist as pro-only UCI options (`BackendPolicyBlendOptimistic`, `BackendPolicyBlendSoft`) so a tuner can move them as ordinary parameters. They are hidden to pro options as the changes only can apply once the backend constructed. Weights are renormalised onto the simplex if they sum past one, so the blend stays a weighted mean rather than becoming a temperature. A network that lacks the requested head refuses to load, rather than silently not blending.

With no weight set, nothing changes: no extra head is built, no extra buffer is allocated, and the priors are bit-identical to master. That was checked against a build of this branch point over 100 positions, at defaults and under a tuned configuration, with identical bestmoves and no output of any kind. Cost ~0.5% fewer nodes per second with a 256x10 net when enabled. It iss the extra head's own projections plus the mixing kernel.

Tested at: https://bench-direct.lczero.org/test/1153/ Whether blending helps, and at what weights, depends on the net this is a capability rather than a general strength gain. On the net tested it was worth 20.2 +/- 3.1 Elo over 10000 games at 10+0.1 against the same net unblended.

Cuda backends only for now. The same graph would be three Mul and two Add nodes in the onnx converter, which would cover the onnx and xla backends and their execution providers without a kernel, if this turns out to be worth having.

Developed with the help of Claude agents for research, implementation and verification; all design decisions, testing and review are my own.

### CODE REVIEW

File: src/neural/backends/cuda/layers.cc [hunk 1]
```diff
@@ -1435,8 +1436,15 @@ AttentionPolicyHead<DataType>::AttentionPolicyHead(
   encoder_heads_ = weights.pol_encoder_head_count;
   policy_d_model_ = wq_op_size_;
 
-  allocAndUpload<DataType>(&ip_pol_w_, weights.ip_pol_w, scratch);
-  allocAndUpload<DataType>(&ip_pol_b_, weights.ip_pol_b, scratch);
+  if (shared_embedding_) {
+    // Another head owns the embedding weights and computes the embedding.
+    assert(weights.pol_encoder.empty());
+    ip_pol_w_ = nullptr;
+    ip_pol_b_ = nullptr;
```
[the remark this answers is not in the document]
Comment (Kovax007) [reply src/neural/backends/cuda/layers.cc:1443]: cudaFree(nullprt) on the nullptr assignment is what prevents the double free
